### ISSUE TITLE
Operator Usage Whitespace Violation with Range Operator "..."

### ISSUE THREAD

**Issue by nicolas-miari** [realm/SwiftLint#2344]
### New Issue Checklist

- [✓] Updated SwiftLint to the latest version
- [✓] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Bug Report

I can't get rid of this warning after migrating to Swift4 and adopting rrange operators for substrings (slices).

##### Complete output when running SwiftLint, including the stack trace and command used

(Can't paste complete output; source file names are confidential)

```
warning: Operator Usage Whitespace Violation: Operators should be surrounded by a single whitespace when they are being used. (operator_usage_whitespace)
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 
0.25.1

* Installation method used (Homebrew, CocoaPods, building from source, etc)? 
Homebrew

* Paste your configuration file:

```yml
# Swift 4.1, SwiftLint 0.25.1
reporter: xcode
opt_in_rules:
  - force_unwrapping
  - vertical_whitespace
  - closure_end_indentation
  - closure_spacing
  - explicit_init
  - first_where
  - nimble_operator
  - number_separator
  - object_literal
  - operator_usage_whitespace
  - overridden_super_call
  - prohibited_super_call
  - redundant_nil_coalescing
disabled_rules:
  - xctfail_message
  - trailing_whitespace
  - cyclomatic_complexity
  - todo
excluded:
  - Carthage
  - Pods
  - vendor
  - FRCircleTests
cyclomatic_complexity:
  warning: 17
line_length: 200
file_length: 900
force_unwrapping: error
function_parameter_count:
  - 6
  - 8
function_body_length:
  - 80
  - 100
type_body_length:
  - 400
  - 500
type_name:
  min_length:
    warning: 2
  max_length:
    warning: 40
    error: 60
identifier_name:
  min_length:
    warning: 1
  max_length:
    warning: 40
    error: 60
identifier_name:
  min_length:
    warning: 1
  max_length:
    warning: 40
    error: 60
custom_rules:
  disable:
    name: "Disable Rule In Code"
    regex: "(swiftlint:disable)"
    message: "Please do not disable rule in code. "
    severity: error
    match_kinds: comment
  outlet_collection:
    name: "outlet_collection"
    regex: "(@IBOutletCollection)"
    message: "IBOutletCollection is not permitted. "
    severity: error
  comments_space:
    name: "Add Space After Comment"
    regex: "(^\\s*//\\w+)"
    message: "There should be a space after //"
    severity: warning
    match_kinds: comment

```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.
No.

* Which Xcode version are you using (check `xcode-select -p`)?
9.4.1

* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
let length = text.length
let startIndex = text.index(text.startIndex, offsetBy: length - 1)

// This does NOT trigger a violation:
let first = String(text[..<startIndex]) 
// Was: text.substring(to: startIndex) in Swift3

// This triggers a violation:
let last =  String(text[startIndex...])  
// Was: text.substring(from: startIndex) in Swift3
```

**Comment by nicolas-miari** [comment]
(Of course, adding space around `...` triggers a Swift syntax error)

**Comment by marcelofabri** [comment]
@nicolas-miari are you sure the violation is not because the extra space between `=` and `String`?

**Comment by marcelofabri** [comment]
Changed here to `let last = String(text[startIndex...])` and the violation went away. I'm closing this, but feel free to reopen if you have any other questions.
